### PR TITLE
APP-4520 feat(dropdown): add size and variant prop

### PR DESCRIPTION
**Pull request by AliciaSymphony**
## Ticket
https://perzoinc.atlassian.net/browse/APP-4520

## Description 
- Add size prop
- Add destructive prop
- Adjust keyboard navigation to the hint

## Comment
- I am having issue implementing size: small, the height should be 24px, but it is currently 32px (as medium) (--> not proposing small size for now) https://perzoinc.atlassian.net/browse/APP-4689

## Demo
image

image

image

Styles: https://github.com/SymphonyPlatformSolutions/symphony-bdk-ui-toolkit-styles/pull/176

### CODE REVIEW

File: src/components/dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -262,6 +257,14 @@ export class Dropdown<T = LabelValue> extends React.Component<
           menuShouldBlockScroll={menuShouldBlockScroll}
           menuShouldScrollIntoView={menuShouldScrollIntoView}
         />
+        <LabelTooltipDecorator
```
Comment (symphony-adnane) [inline src/components/dropdown/Dropdown.tsx:260]: is it normal we move this to the bottom ?

**Review by symphony-adnane — APPROVED** [review]

File: src/components/dropdown/CustomRender.tsx [hunk 2]
```diff
@@ -81,21 +81,20 @@ export const DefaultTagRenderer = (props: any) => {
...
         <TagRender {...rendererProps} />
       </div>
       : <components.MultiValue {...props}>
-        <div onMouseDown={stopPropagation} className="tk-px-1" style={{display:'flex'} }>
-          <div className="tk-tag tk-pr-1">
+        <div onMouseDown={stopPropagation} style={{display:'flex', alignItems: 'center'} }>
+          <div className="tk-tag">
             {props.data?.label}
           </div>
-          <div>
```
Comment (symphony-jean-michael) [inline src/components/dropdown/CustomRender.tsx:93]: 👍

File: src/components/dropdown/CustomRender.tsx [hunk 3]
```diff
@@ -132,8 +131,9 @@ export const ClearIndicator = (props: any) =>
 
 export const Control = ({ children, selectProps, ...props }: any) => {
   const {iconName} = selectProps;
-  return (<div className="tk-input-group__header tk-pt-h">
-    <components.Control {...props} className="tk-input__container">
+  return (<div>
```
Comment (symphony-jean-michael) [inline src/components/dropdown/CustomRender.tsx:134]: Is this 'div' still necessary?

File: src/components/dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -151,6 +153,11 @@ export class Dropdown<T = LabelValue> extends React.Component<
       maxHeight,
       maxMenuHeight,
       menuIsOpen,
+      menuPlacement,
```
Comment (symphony-jean-michael) [inline src/components/dropdown/Dropdown.tsx:156]: Cool to re-order props 👍

File: src/components/dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -162,32 +169,20 @@ export class Dropdown<T = LabelValue> extends React.Component<
...
-      <div className={className}>
-        <LabelTooltipDecorator
-          htmlFor={id}
-          label={label}
-          placement={'top'}
-          tooltip={tooltip}
-          tooltipCloseLabel={tooltipCloseLabel}
-          showRequired={showRequired}
-        />
+      <div className={classNames(className, `${prefix}-group`)}>
```
Comment (symphony-annaLlorens) [inline src/components/dropdown/Dropdown.tsx:185]: What's the group?

File: src/components/dropdown/interfaces.ts [hunk 6]
```diff
@@ -164,6 +166,8 @@ export type DropdownProps<T> = {
     | React.FunctionComponent<TagRendererProps<T>>;
   /** Message to be display on the header of the menu list when searching by term */
   termSearchMessage?: ((term: string) => string) | string;
+  /** Color variant of the button*/
```
Comment (symphony-annaLlorens) [inline src/components/dropdown/interfaces.ts:169]: /** Color variant of the dropdown*/

**Review by symphony-jean-michael — APPROVED** [review]

File: src/components/dropdown/CustomRender.tsx (hunk 3 above)
Comment (AliciaSymphony) [reply src/components/dropdown/CustomRender.tsx:134]: Yes, otherwise the styles break

File: src/components/dropdown/Dropdown.tsx (hunk 5 above)
Comment (AliciaSymphony) [reply src/components/dropdown/Dropdown.tsx:185]: le group is the wrapper of the dropdown and the label

File: src/components/dropdown/Dropdown.tsx (hunk 1 above)
Comment (AliciaSymphony) [reply src/components/dropdown/Dropdown.tsx:260]: Yes... for keyboard navigation (but kept visually at the top with `order: -1`)

File: src/components/dropdown/interfaces.ts (hunk 6 above)
Comment (AliciaSymphony) [reply src/components/dropdown/interfaces.ts:169]: oopsi

**Pull request merged**